packages/babel-parser/src/parser/expression.js

```diff
@@ -766,6 +766,7 @@ export default class ExpressionParser extends LValParser {
     optional: boolean,
   ): N.Expression {
     const oldMaybeInArrowParameters = this.state.maybeInArrowParameters;
+    const refExpressionErrors = new ExpressionErrors();
```

Comment:
I think we should create `refExpressionErrors` only when `state.maybeAsyncArrow` is `true`, otherwise `refExpressionErrors !== undefined` is no longer equivalent to `possibleAsyncArrow`.

Reply:
I push this update in the commit `7ab3018`. Let me know I got you right.